### ISSUE TITLE
Running only a subset of member

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#799]
In GitLab by @acarreri on Mar 16, 2022, 15:00

Hi @dbeltrankyl ,

I'm using the command `autosubmit run a4fq -rm "fc00 fc01"` with a seasonal forecast. 
And autosubmit told me that it was doing what I asked with the corresponding line:
~~~
Only jobs with member value in ['fc00', 'fc01', 'fc02', 'fc03'] or no member will be allowed in this run. Also, those jobs already SUBMITTED, QUEUING, or RUNNING will be allowed to complete and will be tracked.
~~~
 
It was working as expected, launching only the first two members.

But today, I realised that for the first startdate `19901101` of this experiment `a4fq`, a lot of other members have been launched. I stopped AS and relaunched it adding two members, hoping that it will stop launching all members by startdate, but in any case, for this startdate `19901101`, the jobs are already queuing and I should cancel it myself.

Note that I didn't really stop AS until now after the first command to run 2 members. when I had some failed jobs, I relaunched them via the update.txt files in the pkl folder. I don't know if it could be related?   

Could you have a look at it? AS should stop launching the jobs after fc00 and fc01, right?

Thanks

FYI @eferre1 @mcastril

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 16, 2022, 17:23

> But today, I realised that for the first startdate `19901101` of this experiment `a4fq`, a lot of other members have been launched. I stopped AS and relaunched it adding two members, hoping that it will stop launching all members by startdate, but in any case, for this startdate `19901101`, the jobs are already queuing and I should cancel it myself.

@acarreri you say that it's working well for some startdates but not for this one?

**Comment by kinow** [comment]
In GitLab by @acarreri on Mar 16, 2022, 17:33

When the experiment reached the point when the job for the last startdate 2015 (and the two first members) were launching (queuning at that time), it then launched several other members for the first startdate, even if I haven't asked that with my first AS command line.

But I just checked, and stoping and relaunching again the AS command line `autosubmit run a4fq -rm "fc00 fc01 fc02 fc03"` seems to have worked as now, only jobs for this members are queuing for the first 6 startdates (from 1990 to 1995). The thing is that if I haven't checked the exp, all the jobs for the first startdate would have been launched. 

I hope I'm clear enough! Not so easy to describe.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 17, 2022, 09:42

Hello, Maybe I miss something because this feature was coded by @wuruchi . 

While looking at the code, seems that the job_list is set from the beginning. If there are (external) updates to the job_list during the running of the workflow may some issues appear.

Like:

> when I had some failed jobs, I relaunched them via the update.txt

So the recommendation is not update the failed jobs without stop the experiment in this case until I do a more deep research

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 17, 2022, 18:59

Good point. But this shouldn't affect jobs that have not been launched yet and therefore have not had the possibility to fail....

**Comment by kinow** [comment]
In GitLab by @acarreri on Apr 4, 2022, 11:46

Hi,

Just to tell you that stopping autosubmit (and cancel in mn4 the jobs that was launched without I ask for them, that is the other members) and running again autosubmit with the command `autosubmit run a4fq -rm "fc00 fc01 fc02 fc03 fc04 fc05 fc06 fc07 fc08 fc09"`, it launched other members not asked.

You can see on the gui https://earth.bsc.es/autosubmitapp/experiment/a4fq that for the first startdates, other members have run during the week-end. When for the last startdates, the members asked have not been launching yet. 

So, even when I don't update the status with the pkl file, this command doesn't work well.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Apr 5, 2022, 11:27

Thanks for the feedback, 

I'll look into this today

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Apr 6, 2022, 17:01

I put a possible fix into 3.14.0b version. 

I'll add it into 3.13.0 tomorrow if everything is fine.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Apr 8, 2022, 17:47

Hello, 

The fix is working on 3.14.0b. But since I have holidays until 21th I will delay the 3.13.0 fix until then ( to avoid unexpected issues while I'm not available)

**Comment by kinow** [comment]
In GitLab by @acarreri on May 11, 2022, 16:17

mentioned in issue Experiments#251
